lib/rules/id-length.js

```diff
@@ -83,6 +84,13 @@ module.exports = {
                 return parent.id === node;
             },
             Property: properties && function(parent, node) {
+
+                if (parent.parent.type === "ObjectPattern") {
+                    return (
+                        parent.value !== parent.key && parent.value === node ||
+                        parent.value === parent.key && parent.key === node
+                    );
+                }
                 return parent.key === node;
```

Comment:
This retains the actual behavior, and that is to entirely ignore all object expression and object pattern properties when `"properties"` option is set to `"never"`, but we should probably revisit that at some point. 

The intention of this option is, most likely, to allow creating properties with invalid names. But, it also allows this:

```js
/*eslint id-length: ["error", { "min": 2, "properties": "never" }]*/

const { a } = obj; // no error, this is likely a false negative

const { prop: b } = obj; // no error, this is even more likely a false negative
```

Additionally, and unrelated to the previous, this is probably a bug:

```js
/*eslint id-length: ["error", { "min": 2, "properties": "always" }]*/

const foo = { [a]: 1 }; // error on a, this is likely a false positive

const bar = { [a + b]: 1 }; // no error
```

Reply:
@mdjermanovic 
Thanks 👍 for the review. I think both can be fixed. I'll try : )